src/v/resource_mgmt/memory_groups.cc

```diff
@@ -43,11 +43,31 @@ struct memory_shares {
     }
 };
 
+// Computes the compaction memory to reserve, based on configs and amount of
+// reservable memory (i.e. not use by other subsystems like WASM).
+size_t compaction_reservation(size_t reservable_memory) {
+    const auto& cluster_cfg = config::shard_local_cfg();
+    if (!cluster_cfg.log_compaction_use_sliding_window.value()) {
+        return 0;
+    }
+    auto pct_limit_opt
+      = cluster_cfg.storage_compaction_key_map_memory_limit_percent.value();
+    auto bytes_cfg = cluster_cfg.storage_compaction_key_map_memory.value();
+    if (!pct_limit_opt.has_value()) {
+        return bytes_cfg;
+    }
+    size_t bytes_limit = reservable_memory * pct_limit_opt.value();
```

Comment:
isn't `storage_compaction_key_map_memory_limit_percent` a double on the scale from `1-100` so this will use 12 times the reservable memory by default? Don't we need to divide that by 100?

Reply:
Oh my.... 🙊

Reply:
Fixed, and rejiggered the new code structure to be more testable